src/ApplicationLogs/LogReader.cs

```diff
@@ -176,8 +176,13 @@ private void OnCommitting(NeoSystem system, Block block, DataCache snapshot, IRe
             foreach (var appExec in applicationExecutedList.Where(p => p.Transaction != null))
             {
                 var txJson = TxLogToJson(appExec);
+                if (RpcServer.LogEvents.TryGetValue(appExec.Transaction.Hash, out var list))
+                {
+                    txJson["logs"] = new JArray(list.Select(q => new JString(q.Message)));
+                }
                 Put(appExec.Transaction.Hash.ToArray(), Neo.Utility.StrictUTF8.GetBytes(txJson.ToString()));
             }
+            RpcServer.LogEvents.Clear();
```

Comment:
`Clear()` has been called in RpcServer.

Reply:
This is to make sure logs being removed from the memory as soon as possible.